abdiff/core/create_final_records.py

```diff
@@ -38,13 +37,10 @@ def create_final_records(
     """
     logger.info("Creating final records dataset from 'diffs' and 'metrics' datasets.")
     run_data = read_run_json(run_directory)
-
-    diffs_dataset = load_dataset(diffs_dataset_path)
-    metrics_dataset = load_dataset(metrics_dataset_path)
-
     metrics_timdex_field_columns = run_data["metrics"]["summary"]["fields_with_diffs"]
 
     # get list of unique columns from metrics dataset, and create final dataset schema
+    metrics_dataset = load_dataset(metrics_dataset_path)
```

Comment:
Will this line load the full `metrics_dataset` into memory?

Reply:
This is using our helper function which _lazily_ loads the dataset, so data is not pulled into memory but we can access the schema of the dataset.